vendors/sp1/src/baby_bear_poseidon2.rs

```diff
@@ -78,6 +80,9 @@ pub struct DuplexChallenger<F, P, const WIDTH: usize, const RATE: usize> {
     _marker: PhantomData<(F, P)>,
 }
 
+unsafe impl Send for ffi::DuplexChallenger {}
+unsafe impl Sync for ffi::DuplexChallenger {}
```

Comment:
This was needed because

```rust
impl<Val, Dft, InputMmcs, FriMmcs, Challenge, Challenger> Pcs<Challenge, Challenger>
    for TwoAdicFriPcs<Val, Dft, InputMmcs, FriMmcs>
where
    // ...
    Challenger: CanObserve<FriMmcs::Commitment>
        + CanSample<Challenge>
        + GrindingChallenger<Witness = Val>
        + HasCXXDuplexChallenger,
```

and `GrindingChallenger` trait needs to implement `Sync` trait, but `Send` trait is not needed, I'll remove it and add a comment on `Sync` trait